### PR TITLE
Use v3.1 imagehash.hex_to_hash

### PR DESCRIPTION
**Pull request by bjlittle**
Now that https://github.com/SciTools/conda-recipes-scitools/pull/210 has been merged (thanks to @ocefpaf in https://github.com/SciTools/conda-recipes-scitools/pull/211) we can now use the shiny new `v3.1` of `imagehash` which has the fix required to `imagehash.hex_to_hash` for variable `hash_size`'s.

### CODE REVIEW

**Comment by bjlittle** [comment]
We've got some looooong running tests, that are starting to kill us ... something reeks within.

File: lib/iris/tests/__init__.py [hunk 1]
```diff
@@ -776,7 +761,9 @@ def _hex_to_hash(hexstr, hash_size=_HASH_SIZE):
             else:
                 uris = repo[unique_id]
                 # Create the expected perceptual image hashes from the uris.
-                expected = [_hex_to_hash(os.path.splitext(os.path.basename(uri))[0])
+                to_hash = imagehash.hex_to_hash
```
Comment (dkillick) [inline lib/iris/tests/__init__.py]: Bit odd... does this keep the next line <80char in length?

File: lib/iris/tests/idiff.py [hunk 2]
```diff
@@ -137,25 +137,11 @@ def skip(event):
...
-        h = hexstr[i*2:i*2+2]
-        v = int("0x" + h, 16)
-        l.append([v & 2**i > 0 for i in range(8)])
-    return imagehash.ImageHash(np.array(l))
-
-
 def _calculate_hit(uris, phash, action):
     # Create the expected perceptual image hashes from the uris.
-    expected = [_hex_to_hash(os.path.splitext(os.path.basename(uri))[0])
+    to_hash = imagehash.hex_to_hash
```
Comment (dkillick) [inline lib/iris/tests/idiff.py]: Again

File: .travis.yml [hunk 3]
```diff
@@ -107,7 +107,7 @@ install:
 
 script:
   - if [[ $TEST_TARGET == 'default' ]]; then
-      python -m iris.tests.runner --default-tests --system-tests --print-failed-images;
+      travis_wait python -m iris.tests.runner --default-tests --system-tests --print-failed-images;
```
Comment (dkillick) [inline .travis.yml]: Does this change belong here..?

File: lib/iris/tests/__init__.py (hunk 1 above)
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: Yes

File: lib/iris/tests/idiff.py (hunk 2 above)
Comment (bjlittle) [inline lib/iris/tests/idiff.py]: Yes, again :wink:

**Comment by QuLogic** [comment]
Should just strip those last two...

File: .travis.yml (hunk 3 above)
Comment (bjlittle) [inline .travis.yml]: Just testing this out ...

File: lib/iris/tests/runner/_runner.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def run(self):
 
         args = ['', None, '--processes=%s' % n_processors,
                 '--verbosity=2', regexp_pat,
-                '--process-timeout=600']
+                '--process-timeout=180']
```
Comment (bjlittle) [inline lib/iris/tests/runner/_runner.py]: @dkillick This change brought the `TEST_TARGET=default` matrix runners from `+18m` down to `6m17s` for `py27`, and `7m48s` for `py34` ... that's at least **10 minutes** quicker than I've been seeing for a while ... :tada:

Comment (dkillick) [inline lib/iris/tests/runner/_runner.py]: This number sounds suspiciously like the default image tolerance from the image tests... Still, we can change the value as we need, and always put some effort into fixing it in a year or two :wink: 

And it's great to get the travis tests running faster! Good stuff @bjlittle :+1:

**Pull request merged**